File: Dummy/hawkshaw/library/ktextensions/ConnectivityManagerKt.java
package com.hawkshaw.library.ktextensions;

import Y1.K;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public final class ConnectivityManagerKt {
    public static final boolean isOnMobileData(ConnectivityManager connectivityManager) {
        K.n(connectivityManager, "<this>");
        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(0);
        return networkInfo != null && networkInfo.isConnectedOrConnecting();
    }

    public static final boolean isOnWifi(ConnectivityManager connectivityManager) {
        K.n(connectivityManager, "<this>");
        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(1);
        return networkInfo != null && networkInfo.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/hawkshaw/library/ktextensions/ConnectivityManagerKt.kt
package com.hawkshaw.library.ktextensions

import android.net.ConnectivityManager

/**
 * Utility functions for ConnectivityManager
 */

/**
 * Checks if the device is currently using mobile data
 */
fun ConnectivityManager.isOnMobileData(): Boolean {
    val networkInfo = this.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
    return networkInfo != null && networkInfo.isConnectedOrConnecting
}

/**
 * Checks if the device is currently using WiFi
 */
fun ConnectivityManager.isOnWifi(): Boolean {
    val networkInfo = this.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
    return networkInfo != null && networkInfo.isConnectedOrConnecting
} 